src/proteus/dswx_hls.py

```diff
@@ -3437,47 +3670,61 @@ def generate_dswx_layers(input_list,
                           description=band_description_dict['WTR'],
                           scratch_dir=scratch_dir,
                           output_files_list=build_vrt_list)
-
-    # Output the WTR layer as the browse image
+    
+    # Output the WTR-2 w/ Translucent clouds layer as the browse image
     # Note: The browse image will be always be saved as a separate png file;
     # it will not included in the combined `output_file`.
     if output_browse_image:
 
-        # If the `output_interpreted_band` was generated,
-        # convert that to the browse image
-        if output_interpreted_band:
-            src_geotiff_filename = output_interpreted_band
-        else:
-            # If the `output_interpreted_band` geotiff was not generated,
-            # we'll need to make it as a temporary file, and then convert
-            # that to the browse image.
-
-            # Create the source image as a geotiff
-            # Reason: gdal.Create() cannot currently create .png files, so we
-            # must start from a GeoTiff, etc.
-            # Source: https://gis.stackexchange.com/questions/132298/gdal-c-api-how-to-create-png-or-jpeg-from-scratch
-            src_geotiff_filename = os.path.join(scratch_dir,"dswx_tmp.tif")
-            save_dswx_product(masked_dswx_band,
-                            src_geotiff_filename,
-                            dswx_metadata_dict,
-                            geotransform,
-                            projection,
-                            scratch_dir=scratch_dir)
-
-            # add the temp file to the list to be removed at the end
-            temp_files_list += [src_geotiff_filename]
-
-        # Convert the geotiff to a PNG to create the browse image
-        geotiff2png(src_geotiff_filename=src_geotiff_filename,
+        # Build the browse image
+        # Create the source image as a geotiff
+        # Reason: gdal.Create() cannot currently create .png files, so we
+        # must start from a GeoTiff, etc.
+        # Source: https://gis.stackexchange.com/questions/132298/gdal-c-api-how-to-create-png-or-jpeg-from-scratch
+        wtr2_with_translucent_clouds = _compute_wtr2_with_translucent_clouds_array(
+                    unmasked_interpreted_water_layer=landcover_shadow_masked_dswx,  # WTR-2
+                    cloud_layer=cloud,
+                    flag_collapse_wtr_classes=FLAG_COLLAPSE_WTR_CLASSES,
+                    flag_no_snow_mask=flag_no_snow_mask_browse)
```

Comment:
These lines are way too wide. This is not a strict rule, but please try to fit them within 80 chars.